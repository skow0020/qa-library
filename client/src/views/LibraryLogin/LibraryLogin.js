import AlertModal, { showAlert } from 'components/common/AlertModal';
import React, { useState } from 'react';

import Avatar from '@material-ui/core/Avatar';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import Colors from 'utils/Colors';
import Container from '@material-ui/core/Container';
import Copyright from 'components/common/Copyright';
import CssBaseline from '@material-ui/core/CssBaseline';
import Link from '@material-ui/core/Link';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import { Redirect } from 'react-router-dom';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: Colors.primary
  },
  form: {
    width: '100%',
    marginTop: theme.spacing(1)
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
    backgroundColor: Colors.primary,
    color: Colors.white
  }
}));

export default function SignIn() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [authenticated, setAuthenticated] = useState(false);

  function onSubmit(event) {
    event.preventDefault();
    fetch('/api/users/authenticate', {
      method: 'POST',
      body: JSON.stringify({ email, password }),
      headers: { 'Content-Type': 'application/json' }
    })
      .then(response => response.json())
      .then(res => {
        if (res.success) setAuthenticated(true);
        else showAlert({ message: `Error logging in: ${res.error}` });
      });
  }

  const classes = useStyles();
  if (authenticated) return <Redirect to='/qa-dashboard' />;
  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <AlertModal />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component="h1" variant="h5">
          Library Sign in
        </Typography>
        <form className={classes.form} onSubmit={onSubmit}>
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            id="email"
            label="Email Address"
            name="email"
            autoComplete="email"
            autoFocus
            onChange={(e) => { setEmail(e.target.value); }}
          />
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            name="password"
            label="Password"
            type="password"
            id="password"
            autoComplete="current-password"
            onChange={(e) => { setPassword(e.target.value); }}
          />
          <Button
            type="submit"
            fullWidth
            variant="contained"
            id="submit-button"
            className={classes.submit}
          >
            Sign In
          </Button>          
        </form>
        <Link id="registration-link" href="/registration" variant="body2">
            {'Don\'t have an account? Sign Up'}
        </Link>
      </div>
      <Box mt={8}>
        <Copyright />
      </Box>
    </Container>
  );
}
