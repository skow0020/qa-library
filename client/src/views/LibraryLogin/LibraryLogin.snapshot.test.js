import LibraryLogin from './LibraryLogin';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { createSerializer } from 'enzyme-to-json';
import { mount } from 'enzyme';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('LibraryLogin Snapshot Tests', () => {
  test('LibraryLogin snapshot', () => {
    const wrapper = mount(
      <Router>
        <LibraryLogin />
      </Router>
    );

    expect(wrapper).toMatchSnapshot();
  });
});