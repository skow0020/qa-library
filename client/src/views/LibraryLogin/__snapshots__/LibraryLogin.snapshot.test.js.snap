// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibraryLogin Snapshot Tests LibraryLogin snapshot 1`] = `
<div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div />
    <div
      class="makeStyles-paper-1"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Library Sign in
      </h1>
      <form
        class="makeStyles-form-3"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
            data-shrink="true"
            for="email"
            id="email-label"
          >
            Email Address
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="email"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="email"
              name="email"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-124 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-126 PrivateNotchedOutline-legendNotched-127"
              >
                <span>
                  Email Address
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink="false"
            for="password"
            id="password-label"
          >
            Password
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="current-password"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-124 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-126"
              >
                <span>
                  Password
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-fullWidth"
          id="submit-button"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Sign In
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
          href="/registration"
          id="registration-link"
        >
          Don't have an account? Sign Up
        </a>
      </form>
    </div>
    <div
      class="MuiBox-root MuiBox-root-167"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Copyright © 
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="https://github.com/skow0020"
        >
          Colin Skow
        </a>
         
        2020
        .
      </p>
    </div>
  </main>
</div>
`;
